### ISSUE TITLE
First login window closing not detected on macos

### ISSUE THREAD

**Issue by gladiatortoise** [gladiatortoise/node-apiless-youtube-upload#6]
@ankitsiliconithub encountered this issue where first login window closing is not detected, and therefore the program just halts and waits for something to happen.
Interstingly, none of the following event handlers fire on closing:
```
chromeProcess.on('error', cancel);
chromeProcess.on('close', success);
chromeProcess.on('exit', () => console.log('process exited 117'));
chromeProcess.on('disconnect', () => console.log('process disconnected 118'));
```

I will debug this more on a macos machine later.

**Comment by evilive3000** [comment]
On Linux closing the window fires the 'close' event.

But I still can not upload the video. 

On the next steps when we trying to open the youtube studio I got redirected to the `account.google.com`, seems google does not like the cookies we stored a moment before.
```js
  await driver.get(YOUTUBE_STUDIO_URL)
```

so I got this message eventually:
```
"The login session could not be loaded (either user never logged in, random lag or google account doesn't have a youtube attached)",
```

**Comment by gladiatortoise** [comment]
@evilive3000 
It's weird because I thought I tackled that issue with first sleeping 4 sec, then going to google.com and only after that to youtube.studio.com

```typescript
await driver.get('https://www.portofastoria.com/pleasewait.htm');
await driver.sleep(4000)
await driver.get('https://google.com');
// Open youtube studio to test if the login is valid
await driver.get(YOUTUBE_STUDIO_URL);
```
I wonder if I should just 2x the sleep :D

**Comment by evilive3000** [comment]
I just tested with 4x sleep (16 seconds) delay, nothing changed.

after 'portofastoria' I was redirected to the accounts.google.com with the SignIn form.
But even if I signing in manually, I get the message "This browser or app may not be secure".

What is more in the opened window I see the bar saying "Chrome is being controlled by automated test software", I think this is the reason for the message about "insecure app"

**Comment by gladiatortoise** [comment]
Right, the "This browser or app may not be secure" is expected. The reason the sign in is done in two steps, is because when you enter credentials, the chrome must not be controlled by selenium.

The problem here is that the logged in profile doesn't for some reason fully load in the controlled chrome. Just a quick question, are you sure you compiled the typescript after editing, or edited directly the dist/[x].js file?

**Comment by evilive3000** [comment]
I used `ts-node`.
And I did a lot of other debugging stuff, so it's not a compilation problem.
Will try to ensure the cookie is loaded now.
But to make it clear, can you confirm that the script works for you as of today?

**Comment by gladiatortoise** [comment]
@evilive3000 
Yes, just now on a Windows 10 I ran `ts-node usage.ts`, logged in and uploaded succesfully

**Comment by evilive3000** [comment]
Great, this is encouraging. I'll let you know if I figure out my problem.
Anyway thanks for the work you've done.

Would you mind if I might do some PR later?

**Comment by gladiatortoise** [comment]
One idea would be to first make sure that the logged in profile is acknowledged by the controlled brower by going to chrome://version in a new tab. It should show the start  flags, which should include --user-data-dir
![image](https://user-images.githubusercontent.com/5674770/106368014-22a90300-634f-11eb-80bd-a5fb1959af25.png)
If there's some problem with the path for example, chrome tends to just not even show it there.



> Great, this is encouraging. I'll let you know if I figure out my problem.
> Anyway thanks for the work you've done.
> 
> Would you mind if I might do some PR later?

Sure man! I'll take a look at this myself too later on a VM if you don't have time

**Comment by evilive3000** [comment]
I copied the folder created by the method `makeLoggedInChromeProfile`;
Then I tried to rerun this method and fed copied folder into it, so this time chrome instance starts with the previously logged profile.
It worked as expected. The opened window was correctly signed with my user in google services.

Then I tried to feed this folder to the controlled instance of the browser and it's not worked. 
So the cookies are stored correctly but the controlled version of chrome ignore it :(

**Comment by gladiatortoise** [comment]
Ok weird.
I'll see today if I can replicate this with Ubuntu. Which distro are you using?

**Comment by evilive3000** [comment]
`Ubuntu 18.04.5 LTS`

**Comment by gladiatortoise** [comment]
Hi! I got my Ubuntu VM running. I started removing chrome startup flags one by one and found the suspect. It's ```--password-store=basic``` flag. Removing that fixes the issue. I'll build the new version in a bit.

**Comment by gladiatortoise** [comment]
Hi! I got my Ubuntu VM running. I started removing chrome startup flags one by one and found the reason. It's the ```--password-store=basic``` flag. Removing that fixes the issue. I'll build the new version in a bit.

**Comment by gladiatortoise** [comment]
@evilive3000 
```npm install node-apiless-youtube-upload@1.1.8``` should work now

**Comment by evilive3000** [comment]
Great! Now it works!
Thanks

**Comment by matsmello** [comment]
I have the exact same error using this library on macOS and I didn't solve it. Using the new version of the package. Do you have any updates?

**Comment by gladiatortoise** [comment]
> I have the exact same error using this library on macOS and I didn't solve it. Using the new version of the package. Do you have any updates?

This thread is filled with discussion on an unrelated issue. The original issue that you are experiencing is still open, but I will probably solve it once I get my MacOS VM running.

**Comment by matsmello** [comment]
> > I have the exact same error using this library on macOS and I didn't solve it. Using the new version of the package. Do you have any updates?
> 
> This thread is filled with discussion on an unrelated issue. The original issue that you are experiencing is still open, but I will probably solve it once I get my MacOS VM running.

Can I do something to help you?

**Comment by gladiatortoise** [comment]
> > > I have the exact same error using this library on macOS and I didn't solve it. Using the new version of the package. Do you have any updates?
> > 
> > 
> > This thread is filled with discussion on an unrelated issue. The original issue that you are experiencing is still open, but I will probably solve it once I get my MacOS VM running.
> 
> Can I do something to help you?

Could you give me your OS version and Node.js version?

**Comment by matsmello** [comment]
> > > > I have the exact same error using this library on macOS and I didn't solve it. Using the new version of the package. Do you have any updates?
> > > 
> > > 
> > > This thread is filled with discussion on an unrelated issue. The original issue that you are experiencing is still open, but I will probably solve it once I get my MacOS VM running.
> > 
> > 
> > Can I do something to help you?
> 
> Could you give me your OS version and Node.js version?

Node version: v12.18.3
Mac: macOS Big Sur Version 11.1

**Comment by gladiatortoise** [comment]
Thanks. I'll ping you when there's progress

**Comment by matsmello** [comment]
I have some suggestions to add new features and improve this library. After works, is it okay to open a few pull requests?

**Comment by gladiatortoise** [comment]
Sure it is

**Comment by gladiatortoise** [comment]
Apparently on MacOs, when you click the red [X], it doesn't close the process but just hides it so it appears to work smoother when you reopen it. The module doesn't detect it closing because it never closes.

Solution is to "quit" the chrome process
![image](https://user-images.githubusercontent.com/5674770/106675834-7f5e2500-65be-11eb-8ff0-37d9b1b44465.png)

This is of course a bit odd and the user is not expected to know to follow this. This step is not necessary once we implement a way to get window title on MacOs (#7)

Unfortunately the problems don't stop here on MacOs. When getting to the controlled window, there is a similar issue to what was discussed in this thread earlier but for Linux. I'll make a new issue for it
